### PR TITLE
Add UT

### PR DESCRIPTION
**Pull request by mvn-luanvu-hn**
## What I did
Add unit test samples
- Component with react-query
- Component with router
- Component with react-hook-form
- Component with react-i18next
- Component with Zustand store
- Hook
- Context
- Zustand store
- Util

## How to test

- [x] Is this testable with jest or e2e?
- [ ] ~~Does this need an update to the documentation?~~

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: jest-setup.ts [hunk 1]
```diff
@@ -1 +1,25 @@
 import "@testing-library/jest-dom";
+// Need to use whatwg-fetch because mws can not intercept request
+// when using node's built-in fetch
```
Comment (jinmayamashita) [inline jest-setup.ts:3]: 👍🏼

File: src/utils/http.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  test('throws error when calling fetch not successfully', async () => {
+    windowFetchSpy.mockResolvedValueOnce({
+      ok: false,
+    });
+
+    await expect(http.get('/employee/1')).rejects.toThrowError('Server error')
+  })
+
+  test('passes direct url to fetch when url is absolute', async () => {
```
Comment (jinmayamashita) [inline src/utils/http.test.ts:112]: :+1:

File: src/utils/http.ts [hunk 3]
```diff
@@ -16,8 +17,12 @@ const request = async <T>(url: string, opts?: RequestInit): Promise<T> => {
     body: JSON.stringify(opts?.body),
   });
   // TODO handle error
```
Comment (jinmayamashita) [inline src/utils/http.ts:19]:
```suggestion
```
@mvn-luanvu-hn now is it possible to delete this comment?

File: tests/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export { default as render } from '@/tests/utils/render'
+export { default as renderWithQueryClient } from '@/tests/utils/renderWithQueryClient'
```
Comment (jinmayamashita) [inline tests/utils/index.ts:2]: It does not seem necessary to manage in the index file, but what do you think? 🤔

e.g.
```tsx
// src/hooks/useAuth.test.tsx
import { renderWithQueryClient } from '@/tests/utils/renderWithQueryClient'
```

**Review by jinmayamashita — CHANGES_REQUESTED** [review]
@mvn-luanvu-hn 
Thanks for the PR 👍🏼 
I reviewed. Please check some comments.

File: src/utils/http.ts (hunk 3 above)
Comment (mvn-luanvu-hn) [reply src/utils/http.ts:19]: @jinmayamashita I think I will create another PR to handle error in the future
In that PR I will do
- Use toast to show error of 5xx error
- Internationalize error message

Comment (jinmayamashita) [reply src/utils/http.ts:19]: @mvn-luanvu-hn 
Sure. Can you add TODOs such as toast issue to the comments in the code?

File: tests/utils/index.ts (hunk 4 above)
Comment (mvn-luanvu-hn) [reply tests/utils/index.ts:2]: @jinmayamashita At the beginning time of this PR, I used your recommending syntax `import { renderWithQueryClient } from '@/tests/utils/renderWithQueryClient'`
But after writing test for some files, I had to import the render utility again and again.
I think it's better to simplify the import statement by exporting them in the index file.
So developers does not need to type extra renderWithQueryClient or render after `@/tests/utils`
It's good for lazy developers like me 😅

File: src/utils/http.ts (hunk 3 above)
Comment (jinmayamashita) [reply src/utils/http.ts:19]:
```suggestion
  // TODO handle error
  // - Use toast to show error of 5xx error
  // - Internationalize error message
```
@mvn-luanvu-hn I agree. :+1: How about this?

Comment (mvn-luanvu-hn) [reply src/utils/http.ts:19]: @jinmayamashita Yes, I agree with you. I just committed your suggestion.

File: tests/utils/index.ts (hunk 4 above)
Comment (jinmayamashita) [reply tests/utils/index.ts:2]: > So developers does not need to type extra renderWithQueryClient or render after @/tests/utils
It's good for lazy developers like me 😅

@mvn-luanvu-hn 

I am a different kind of lazy man as the number of files under the util folder grows, 
I have a hard time getting the util/index file to update along with it. 😅

But I understand your thoughts 👍🏼

Let's try to manage it in the index file and if another issue comes up, let's discuss it again!

**Review by jinmayamashita — APPROVED** [review]
LGTM 👍🏼

@mvn-luanvu-hn Thanks for your works!
I think you've covered testing for most of our features now!

Screen Shot 2022-10-26 at 12 11 55

**Pull request merged**